group_theory/coset.lean

```diff
@@ -0,0 +1,165 @@
+/-
+Copyright (c) 2018 Mitchell Rowett. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mitchell Rowett, Scott Morrison
+-/
+
+import group_theory.subgroup
+
+open set
+
+universe u
+variable {G : Type u}
+
+def left_coset [has_mul G] (a : G) (S : set G) : set G := image (λ x, a * x) S
+def right_coset [has_mul G] (S : set G) (a : G) : set G := image (λ x, x * a) S
+
+infix ` *l `:70 := left_coset
```

Comment:
we have a lot of ` *x ` notation nowadays, do we really want to fix this to left and right coset? how about changing this to local infix notation?